alf/algorithms/trac_algorithm.py

```diff
@@ -170,21 +175,32 @@ def _update_total_dists(new_action, exp, total_dists):
             return nest_map(lambda x, y: x + y, total_dists, dists)
 
         num_steps = exp_array.step_type.size()
+        # element_shape for `TensorArray` can be (None, ...)
+        batch_size = tf.shape(exp_array.step_type.read(0))[0]
         state = tf.nest.map_structure(lambda x: x.read(0), exp_array.state)
-        batch_size = exp_array.step_type.element_shape[0]
         # exp_array.state is no longer needed
         exp_array = exp_array._replace(state=())
         initial_state = common.zero_tensor_from_nested_spec(
-            self.train_state_spec.actor, batch_size)
+            self.train_state_spec, batch_size)
```

Comment:
The structure of initial_state is train_state_spec. But the state from predict is predict_state_spec. They might be different and cause line 188 to fail.

Reply:
changed to  `rollout` instead of `predict`